Source/WebCore/loader/TextResourceDecoder.cpp

```diff
@@ -662,8 +663,8 @@ String TextResourceDecoder::flush()
 
 String TextResourceDecoder::decodeAndFlush(std::span<const uint8_t> data)
 {
-    String decoded = decode(data);
-    return decoded + flush();
+    auto decoded = decode(data);
+    return makeString(decoded, flush());
```

Comment:
Does `makeString` optimize the case where one argument is a `String` and the other arguments have zero characters? If not, perhaps it should since `flush()` almost always returns the empty string. Or could add a fast path here for the same reason. Is it possible that the `+` operator had that case optimized?